sqlgen/db.go

```diff
@@ -311,7 +311,9 @@ func (db *DB) UpdateRow(ctx context.Context, row interface{}) error {
 		return err
 	}
 
-	if err := db.checkColumnValuesAgainstShardLimit(query.Columns, query.Values); err != nil {
+	if err := db.checkColumnValuesAgainstShardLimit(
+		append(query.Where.Columns, query.Columns...),
```

Comment:
I need to include `Where.Columns` because of the way `MakeUpdateRow()` works in `reflect.go`. `MakeUpdateRow()` adds the `column.Name` and `column.Value` to `where` clause if the `column` is `pk`. As a result, if you shard by `pk` you need to check both the `query.Columns` and `query.Where.Columns`.

Reply:
Just for reference, `WithShardLimit` clause was introduced in:
commit: 
https://github.com/samsarahq/thunder/commit/3a53d564b5b2ebc9b3476f597941cea9db693dfe

PR: https://github.com/samsarahq/thunder/pull/253

Reply:
could this be just the where columns?

Reply:
I was thinking about that but unfortunately I don't think so. For example, in the test ⬇️ the first test case tries to shard with a non pk column. In that case `db.Schema.MakeUpdateRow(row)` adds the column to `column.Where` and we would get the same error.

Are you concerned about perf to append arrays?

Would it just be better to send both arrays and iterate over them instead of concentrating them?